scripts/check_for_update_dependency.py

```diff
@@ -0,0 +1,63 @@
+import re
+import requests
+import toml
+from typing import List
+
+from typing import Optional
+
+
+def get_latest_version(package_name: str) -> Optional[str]:
+    """Fetch the latest version of a package from PyPI."""
+    response = requests.get(f"https://pypi.org/pypi/{package_name}/json")
+    if response.status_code == 200:
+        data = response.json()
+        return str(data["info"]["version"])
+    else:
+        return None
```

Comment:
Why should we fail silently? IMO it would be better to fail with an error, so that the script will get fixed. You can use `response.raise_for_status()` or `assert response.status_code == 200`, for example.